modulemd/include/private/modulemd-util.h

```diff
@@ -63,49 +129,195 @@ G_DEFINE_AUTOPTR_CLEANUP_FUNC (modulemd_tracer, modulemd_trace_free);
+ *
+ * Since: 2.2
+ */
 gboolean
 modulemd_hash_table_sets_are_equal_wrapper (const void *a, const void *b);
 
+/**
+ * modulemd_hash_table_sets_are_equal:
+ * @a: A #GHashTable object.
+ * @b: A #GHashTable object.
+ *
+ * Returns: TRUE if both @a and @b contain an identical set of keys, FALSE if
+ * they differ.
+ *
+ * Only the keys of @a and @b are compared. The values are ignored.
+ *
+ * Since: 2.0
+ */
 gboolean
 modulemd_hash_table_sets_are_equal (GHashTable *a, GHashTable *b);
 
+/**
+ * modulemd_hash_table_equals:
+ * @a: A #GHashTable object.
+ * @b: A #GHashTable object.
+ * @compare_func: A #GEqualFunc function that is called to determine the
+ * equivalence of pairs of #GHashTable values.
+ *
+ * Returns: TRUE if both @a and @b contain identical keys and identical
+ * corresponding values (as determined by @compare_func), FALSE if they
+ * differ.
+ *
+ * Since: 2.2
+ */
 gboolean
 modulemd_hash_table_equals (GHashTable *a,
                             GHashTable *b,
                             GEqualFunc compare_func);
 
+/**
+ * modulemd_strcmp_sort:
+ * @a: A #gconstpointer.
+ * @b: A #gconstpointer.
+ *
+ * Returns: 0 if @a and @b are pointers to identical strings, a negative value
+ * if @a is less than @b, and a positive value if @a is greater than @b.
+ *
+ * Since: 2.0
+ */
 gint
 modulemd_strcmp_sort (gconstpointer a, gconstpointer b);
 
+/**
+ * modulemd_ordered_str_keys:
+ * @htable: A #GHashTable.
+ * @compare_func:  A #GCompareFunc function that is called to determine the
+ * equivalence of pairs of #GHashTable keys from @htable.
+ *
+ * Returns: A #GPtrArray of the keys from @htable sorted according to
+ * @compare_func.
+ *
+ * Since: 2.0
+ */
 GPtrArray *
 modulemd_ordered_str_keys (GHashTable *htable, GCompareFunc compare_func);
 
+/**
+ * modulemd_ordered_str_keys_as_strv:
+ * @htable: A #GHashTable.
+ *
+ * Returns: A #GStrv list of the keys from @htable sorted according to
+ * modulemd_strcmp_sort().
+ *
+ * Since: 2.0
+ */
 GStrv
 modulemd_ordered_str_keys_as_strv (GHashTable *htable);
 
+/**
+ * modulemd_variant_deep_copy:
+ * @variant: A #GVariant opaque data structure.
+ *
+ * Returns: A newly-allocated deep copy of @variant.
+ *
+ * Since: 2.0
+ */
 GVariant *
 modulemd_variant_deep_copy (GVariant *variant);
 
+/**
+ * modulemd_hash_table_unref:
+ * @table: (nullable): A void pointer.
+ *
+ * Decrements the reference count of @table (considered as a #GHashTable). If
+ * the reference count drops to 0, all keys, values, and memory associated with
+ * @table will be released.
+ *
+ * Since: 2.0
+ */
 void
 modulemd_hash_table_unref (void *table);
 
+/**
+ * modulemd_validate_nevra:
+ * @nevra: A NEVRA (Name, Epoch, Version, Release, Architecture) string.
+ *
+ * Returns: TRUE if @nevra is in proper N-E:V-R.A format, FALSE otherwise.
+ *
+ * Since: 2.0
+ */
 gboolean
 modulemd_validate_nevra (const gchar *nevra);
 
 
+/**
+ * MODULEMD_REPLACE_SET:
```

Comment:
Add a note that this helper function is intended for use in `GOBJECT_copy()` functions to simplify copying internal set variables and avoid code duplication.